src/framework/accessibility/internal/accessibilitycontroller.cpp

```diff
@@ -0,0 +1,205 @@
+{
+}
+
+static QAccessibleInterface* muAccessibleFactory(const QString& classname, QObject* object)
+{
+    if (classname == QLatin1String("mu::accessibility::AccessibleObject")) {
+        return static_cast<QAccessibleInterface*>(new AccessibleItemInterface(object));
+    }
+
+    if (classname == QLatin1String("mu::accessibility::AccessibilityController")) {
+        return static_cast<QAccessibleInterface*>(new AccessibleControllerInterface(object));
+    }
+
+    return nullptr;
+}
+
+static void updateHandlerNoop(QAccessibleEvent*)
+{
+}
+
+void AccessibilityController::init()
+{
+    setObjectName("AccessibilityController");
+
+    QAccessible::installFactory(muAccessibleFactory);
+
+    QAccessible::setRootObject(this);
+    QAccessible::installRootObjectHandler([](QObject*) {});
+
+    //! NOTE Disabled any events from Qt
+    QAccessible::installUpdateHandler(updateHandlerNoop);
+}
+
+void AccessibilityController::created(IAccessibility* parent, IAccessibility* item)
+{
+    //! TODO Not working yet
+    parent = nullptr;
+
+    QObject* prnObj = nullptr;
+    if (parent) {
+        prnObj = findItem(parent).object;
+    } else {
+        prnObj = this;
+    }
+
+    Item it;
+    it.parent = parent;
+    it.item = item;
+    it.object = new AccessibleObject(item, prnObj);
+    it.object->setController(shared_from_this());
+    it.iface = QAccessible::queryAccessibleInterface(it.object);
+
+    m_items.append(it);
+
+    LOGI() << "parent: " << (parent ? parent->accessibleName() : "") << ", item: " << item->accessibleName();
+
+    QAccessibleEvent ev(it.object, QAccessible::ObjectCreated);
+    sendEvent(&ev);
+}
+
+void AccessibilityController::destroyed(IAccessibility* aitem)
+{
+    LOGI() << aitem->accessibleName();
+    int idx = indexBy(aitem);
+    IF_ASSERT_FAILED(idx >= 0) {
+        return;
+    }
+    Item item = m_items.takeAt(idx);
+    QAccessibleEvent ev(item.object, QAccessible::ObjectDestroyed);
+    sendEvent(&ev);
+
+    delete item.object;
+}
+
+void AccessibilityController::actived(IAccessibility* aitem, bool isActive)
+{
+    LOGI() << aitem->accessibleName() << " " << isActive;
+    const Item& item = findItem(aitem);
+    IF_ASSERT_FAILED(item.isValid()) {
+        return;
+    }
+
+    QAccessible::State state;
+    state.active = isActive;
+    QAccessibleStateChangeEvent ev(item.object, state);
+    sendEvent(&ev);
+}
+
+void AccessibilityController::focused(IAccessibility* aitem)
+{
+    LOGI() << aitem->accessibleName();
+    const Item& item = findItem(aitem);
+    IF_ASSERT_FAILED(item.isValid()) {
+        return;
+    }
+
+    QAccessibleEvent ev(item.object, QAccessible::Focus);
+    sendEvent(&ev);
+}
+
+void AccessibilityController::sendEvent(QAccessibleEvent* ev)
+{
+    QAccessible::installUpdateHandler(0);
+    QAccessible::updateAccessibility(ev);
+    //! NOTE Disabled any events from Qt
+    QAccessible::installUpdateHandler(updateHandlerNoop);
```

Comment:
isn't it enough to mock it at "init"?

Reply:
In Qt such code
```
void QAccessible::updateAccessibility(QAccessibleEvent* event)
{
        ...
        if (updateHandler) {
            updateHandler(event);
            return;
        }
        ...
}
```

So, I set a noop handler to prevent any events.
To send my event, I reset the handler, call the update event, and set the noop handler again.